src/UnlockEligibility.sol

```diff
@@ -2,253 +2,256 @@
-  {
-    // This module does not deal with standing, so we default to good standing (true)
-    standing = true;
+    /// @inheritdoc IHatsEligibility
+    function getWearerStatus(
+        address _wearer,
+        uint256 /* _hatId */
+    ) public view override returns (bool eligible, bool standing) {
+        // This module does not deal with standing, so we default to good standing (true)
+        standing = true;
 
-    eligible = lock.getHasValidKey(_wearer);
-  }
+        eligible = lock.getHasValidKey(_wearer);
+    }
 
-  /*//////////////////////////////////////////////////////////////
+    /*//////////////////////////////////////////////////////////////
                         UNLOCK HOOK FUNCTIONS
   //////////////////////////////////////////////////////////////*/
 
-  /// @inheritdoc ILockKeyPurchaseHook
-  function keyPurchasePrice(
-    address, /* from */
-    address, /* recipient */
-    address, /* referrer */
-    bytes calldata /* data */
-  ) external view returns (uint256 minKeyPrice) {
-    minKeyPrice = keyPrice;
-  }
-
-  /// @inheritdoc ILockKeyPurchaseHook
-  function onKeyPurchase(
-    uint256, /* tokenId */
-    address, /* from */
-    address recipient,
-    address, /* referrer */
-    bytes calldata, /* data */
-    uint256, /* minKeyPrice */
-    uint256 /* pricePaid */
-  ) external {
-    // caller must be the lock contract
-    _checkIsLock(msg.sender);
-
-    /// @dev Will revert if this contract is not an admin of the hat
-    HATS().mintHat(hatId(), recipient);
-  }
-
-  /// @inheritdoc ILockKeyTransferHook
-  function onKeyTransfer(
-    address, /* lockAddress */
-    uint256, /* tokenId */
-    address, /* operator */
-    address from,
-    address to,
-    uint256 /* expirationTimestamp */
-  ) external {
-    // caller must be the lock contract
-    _checkIsLock(msg.sender);
-
-    /// @dev We use the revoke & mint approach here rather than transfer in case the hat is immutable
-
-    // Revoke the hat from the from address (current wearer) without putting them in bad standing
-    /// @dev Will revert if this contract is the hat's eligibility module
-    HATS().setHatWearerStatus(hatId(), from, false, true);
-
-    // Mint the hat to the to address (new wearer)
-    // Since the key has already been transferred, the new wearer is eligible and so minting will succeed
-    /// @dev Will revert if this contract is not an admin of the hat
-    HATS().mintHat(hatId(), to);
-
-    // QUESTION: or should we disallow transfers?
-  }
-
-  /*//////////////////////////////////////////////////////////////
+    /// @inheritdoc ILockKeyPurchaseHook
+    function keyPurchasePrice(
+        address from,
+        address recipient,
+        address referrer,
+        bytes calldata data
+    ) external view returns (uint256 minKeyPrice) {
+        // Check if referrer fee is correct. Fail minting if incorrect.
+        if (lock.referrerFees(FEE_SPLIT_RECIPIENT) != FEE_SPLIT_PERCENTAGE) {
```

Comment:
I added that. this is away to make sure that if a lock manager removes the fees then the purchase are failing!